src/Microsoft.Azure.WebJobs.Host/Extensions/Tooling.cs

```diff
@@ -0,0 +1,397 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    // Provides additional bookkeeping on extensions 
+    internal class Tooling : ITooling
+    {
+        // Map from binding types to their corresponding attribute. 
+        private readonly IDictionary<string, Type> _attributeTypes = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+
+        // Map of simple assembly name to assembly.
+        private readonly Dictionary<string, Assembly> _resolvedAssemblies = new Dictionary<string, Assembly>(StringComparer.OrdinalIgnoreCase);
+
+        private readonly JobHostConfiguration _config;
+
+        // preferred order of output bindings. 
+        private static Type[] _outTypes = new Type[]
+        {
+            typeof(IAsyncCollector<JObject>),
+            typeof(IAsyncCollector<byte[]>),
+            typeof(IAsyncCollector<string>),
+            typeof(Stream)
+        };
+
+        private static Type[] _inTypes = new Type[]
+        {
+            typeof(JObject),
+            typeof(JArray),
+            typeof(Stream),
+            typeof(string)
+        };
+
+        private IBindingProvider _root;
+
+        public Tooling(JobHostConfiguration config)
+        {
+            _config = config;
+        }
+
+        internal void Init(IBindingProvider root)
+        {
+            this._root = root;
+
+            // Populate assembly resolution from converters.
+            var converter = this._config.GetService<IConverterManager>() as ConverterManager;
+            if (converter != null)
+            {
+                converter.AddAssemblies(this.AddAssembly);
```

Comment:
can you elaborate on the motivation? 

---
In reply to: [108524510](https://github.com/Azure/azure-webjobs-sdk/pull/1093#discussion_r108524510) [](ancestors = 108524510)